dev-python/pytest-qt/pytest-qt-3.3.0.ebuild

```diff
@@ -0,0 +1,37 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7} )
+
+DISTUTILS_USE_SETUPTOOLS=rdepend
+
+inherit distutils-r1 virtualx
+
+DESCRIPTION="pytest plugin for PyQt4 or PyQt5 applications"
+HOMEPAGE="https://pypi.org/project/pytest-qt https://github.com/pytest-dev/pytest-qt"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+DEPEND="dev-python/QtPy[gui,testlib,${PYTHON_USEDEP}]"
+
+PATCHES="${FILESDIR}/${P}-skip-show-window-test.patch"
+
+distutils_enable_tests pytest
+distutils_enable_sphinx docs dev-python/sphinx_rtd_theme
+
+python_test() {
+	pytest_qt_test() {
+		# pytest-qt test fail to test in ${BUILDIR}/lib
+		# if and only if pytest-qt is not already installed
+		# test do work if executed directly in the extracted tarball
+		local PYTHONPATH="${WORKDIR}/${P}"
+		pytest -vv
+	}
+
+	virtx pytest_qt_test || die "Tests fail with ${EPYTHON}"
```

Comment:
Die is probably wrong here... I see the point with your own function, but please remove it for now.

Reply:
Fixed, but doesn't that mean the package will install even if tests fail?

Reply:
I'm not really sure about the interaction with virtx, but the eclass devmanual page says that if you add `|| die` here it will leave the virtual X session open, which is worse imo as it's not something people know what to look for.

Reply:
Alright, I see, we'll do it like this than. Leaving stray X sessions behind does indeed sound like a bad idea.